lib/engine/game/g_1832/game.rb

```diff
@@ -148,8 +148,14 @@ class Game < Game::Base
         CORPORATE_BUY_SHARE_ALLOW_BUY_FROM_PRESIDENT = true
         IPO_RESERVED_NAME = 'Treasury'
 
-        TILE_LAYS = [{ lay: true, upgrade: true, cost: 0, cannot_reuse_same_hex: true },
-                     { lay: :not_if_upgraded, upgrade: false, cost: 0 }].freeze
+        TILE_LAYS = [{ lay: true, upgrade: true }, { lay: :not_if_upgraded, upgrade: false }].freeze
+        SYSTEM_TILE_LAYS = [{ lay: true, upgrade: true },
+                            { lay: :not_if_upgraded, upgrade: false },
+                            { lay: true, upgrade: :not_if_upgraded, cannot_reuse_same_hex: true }].freeze
+
+        def tile_lays(_entity)
+          @entity.system? ? SYSTEM_TILE_LAYS : TILE_LAYS
```

Comment:
```suggestion
          @entity.type == :system ? SYSTEM_TILE_LAYS : TILE_LAYS
```
`entity.system?` is going to be `false` for all corporations, including systems (AFAICT, only 1828’s System class would return `true` for this).

Reply:
How about this: 

```
        def tile_lays(entity)
          return self.class::SYSTEM_TILE_LAYS if system?(entity)

          self.class::TILE_LAYS
        end

        def system?(corporation)
          return false unless corporation

          corporation.type == :system
        end
```
obviously, with type: 'system' in the entities file.

Reply:
Yes, that would work.